File: examples/javascript/presigned-url.mjs
// Hippius S3 — Generate presigned URLs with AWS SDK v3
// Docs: /storage/s3/javascript

import "dotenv/config";
import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";

const s3 = new S3Client({
  endpoint: "https://s3.hippius.com",
  region: "decentralized",
  credentials: {
    accessKeyId: process.env.HIPPIUS_ACCESS_KEY,
    secretAccessKey: process.env.HIPPIUS_SECRET_KEY,
  },
  forcePathStyle: true,
});

const bucketName = "my-bucket";

const url = await getSignedUrl(
  s3,
  new GetObjectCommand({ Bucket: bucketName, Key: "hello.txt" }),
  { expiresIn: 3600 },
);
console.log(`Presigned URL: ${url}`);

File: examples/python/presigned_url.py
# Hippius S3 — Generate presigned URLs with boto3
# Docs: /storage/s3/python

import os
from dotenv import load_dotenv
import boto3
from botocore.config import Config

load_dotenv()

s3 = boto3.client(
    "s3",
    endpoint_url="https://s3.hippius.com",
    aws_access_key_id=os.environ["HIPPIUS_ACCESS_KEY"],
    aws_secret_access_key=os.environ["HIPPIUS_SECRET_KEY"],
    region_name="decentralized",
    config=Config(
        signature_version="s3v4",
        s3={"addressing_style": "path"},
    ),
)

bucket_name = "my-bucket"

url = s3.generate_presigned_url(
    "get_object",
    Params={"Bucket": bucket_name, "Key": "hello.txt"},
    ExpiresIn=3600,
)
print(f"Presigned URL: {url}")
